bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/AbstractBigtableTable.java

```diff
@@ -752,4 +710,69 @@ public void setWriteRpcTimeout(int i) {
   public int getRpcTimeout() {
     throw new UnsupportedOperationException("getRpcTimeout");
   }
+
+  private static Class<? extends AbstractBigtableTable> tableClass = null;
+
+  /**
+   * This is a workaround for incompatible changes in hbase minor versions. Dynamically generates a
+   * class that extends AbstractBigtableTable so incompatible methods won't be accessed unless the
+   * methods are called. If a method is implemented by AbstractBigtableTable, the generated class
+   * will invoke the implementation in AbstractBigtableTable.
+   */
+  private static synchronized Class<? extends AbstractBigtableTable> getSubclass()
```

Comment:
Can we remove the implementation in AbstractBigtableTable? Since it's created in BigtableTable and BigtableAsyncTable

Reply:
Unfortunately no, because we make direct calls to AbstractBigtableTable like here: https://github.com/googleapis/java-bigtable-hbase/blob/main/bigtable-hbase-1.x-parent/bigtable-hbase-1.x/src/main/java/com/google/cloud/bigtable/hbase1_x/BigtableConnection.java#L82

Reply:
(I tried removing it and got errors that some methods were umimplemented)

Reply:
Gotcha!